db/seeds.rb

```diff
@@ -1,187 +1,170 @@
   r.url = "http://www.dot.ri.gov/img/travel/camimages/95-22b%20I-95%20N%20@%20Kinsley%20Ave.jpg"
   r.location = providence
   r.feed_format = "image"
-end.save!
+end
 
-TrafficCam.find_or_initialize_by(
-  title: "Exit 10 (near I-295)"
-) do |r|
+update_or_new(TrafficCam, title: "Exit 10 (near I-295)") do |r|
   r.url = "http://www.dot.ri.gov/img/travel/camimages/95-11%20I-95%20N%20@%20Toll%20Gate%20Rd.jpg"
   r.location = providence
   r.feed_format = "image"
-end.save!
+end
 
-TrafficCam.find_or_initialize_by(
-  title: "I-95 I-195 Split"
-) do |r|
+update_or_new(TrafficCam, title: "I-95 I-195 Split") do |r|
   r.url = "http://www.dot.ri.gov/img/travel/camimages/195-1%20I-195%20W%20Split%20@%20I-95.jpg"
   r.location = providence
   r.feed_format = "image"
-end.save!
+end
 
-TrafficCam.find_or_initialize_by(
-  title: "Dean Street"
-) do |r|
+update_or_new(TrafficCam, title: "Dean Street") do |r|
   r.url = "https://www.dot.ri.gov/img/travel/camimages/6-1%20Rt%206%20E%20%20%2010%20N%20@%20Dean%20St.jpg"
   r.location = providence
   r.feed_format = "image"
-end.save!
+end
 
 TrafficCam.where(
   title: "Tobey Street"
 ).destroy_all
 
-TrafficCam.find_or_initialize_by(
-  title: "Broadway and Canyon"
-) do |r|
+#Seed Boulder traffic cam live-feeds
+
+update_or_new(TrafficCam, title: "Broadway and Canyon") do |r|
   r.url = "https://videostream.bouldercolorado.gov/live/smil:broadway_and_canyon.smil/playlist.m3u8"
   r.location = boulder
   r.feed_format = "video"
-end.save!
+end
 
-TrafficCam.find_or_initialize_by(
-  title: "Foothills and Arapahoe"
-) do |r|
+update_or_new(TrafficCam, title: "Foothills and Arapahoe") do |r|
   r.url = "https://videostream.bouldercolorado.gov/live/smil:foothills_and_arapahoe.smil/playlist.m3u8"
   r.location = boulder
   r.feed_format = "video"
-end.save!
+end
 
-TrafficCam.find_or_initialize_by(
-  title: "28th and Colorado"
-) do |r|
+update_or_new(TrafficCam, title: "28th and Colorado") do |r|
   r.url = "https://videostream.bouldercolorado.gov/live/smil:28th_and_colorado.smil/playlist.m3u8"
   r.location = boulder
   r.feed_format = "video"
-end.save!
+end
 
-TrafficCam.find_or_initialize_by(
-  title: "28th and Iris"
-) do |r|
+update_or_new(TrafficCam, title: "28th and Iris") do |r|
   r.url = "https://videostream.bouldercolorado.gov/live/smil:28th_and_iris.smil/playlist.m3u8"
   r.location = boulder
   r.feed_format = "video"
-end.save!
+end
+
+#Add random events
+
+update_or_new(Event, external_id: "Tucker's final helios PR :(") do |r|
+  r.source = 'github_pull'
+  r.created_at = Date.parse('2021-10-6T10:09:08.07')
+end
+
+update_or_new(Event, external_id: "PR: fix to allow multiple seedings") do |r|
+  r.source = 'github_pull'
+  r.created_at = Date.parse('2021-10-6T10:09:08.07')
+end
+
+update_or_new(Event, external_id: "#the-five-interns-2021 channel message") do |r|
+  r.source = 'slack_message'
+  r.created_at = Date.parse('2021-10-6T10:09:08.07')
+end
```

Comment:
![Screenshot from 2021-08-10 13-47-23](https://user-images.githubusercontent.com/84336508/128914783-70b355f4-a486-41a4-9431-b14a5a271292.png)

Example events appear on helios for the week of August 10th - as expected.

Reply:
These events serve more as tests for the Numbers widget, and their hardcoded dates may be updated by future helios editors to display test events on that day. These test records are not intended to generate current events or artificially increment the events displayed on helios for any given week.
Because they can be edited to test the functionality of the Numbers widget, these test events are created with the update_or_new! method - allowing for successful seeding of their edited fields.